File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int n = nums.size();
        int currSum = nums[0];
        int ans = nums[0];

        for(int i = 1; i < n; i++){
            currSum = max(currSum+nums[i], nums[i]);
            ans = max(ans, currSum);
        }

        return ans;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.go
func maxSubArray(nums []int) int {
    n := len(nums)
    currSum := nums[0]
    ans := nums[0]

    for i := 1; i < n; i++ {
        currSum = max(currSum + nums[i], nums[i])
        ans = max(ans, currSum)
    }

    return ans;
}


func max(a, b int) int{
    if a > b{
        return a
    }
    return b
}